src/code/z_camera.c

```diff
@@ -531,13 +528,13 @@ s32 Camera_GetWaterBoxDataIdx(Camera* camera, f32* waterY) {
         return -1;
     }
 
-    ret = WaterBox_GetCamDataIndex(&camera->play->colCtx, waterBox);
-    if ((ret <= 0) || (WaterBox_GetCameraSType(&camera->play->colCtx, waterBox) <= 0)) {
+    bgCamDataIndex = WaterBox_GetBgCamDataIndex(&camera->play->colCtx, waterBox);
+    if ((bgCamDataIndex <= 0) || (WaterBox_GetCameraSetting(&camera->play->colCtx, waterBox) <= CAM_SET_NONE)) {
         // no camera data index, or no CameraSettingType
```

Comment:
is there a `@bug` here? why is `bgCamDataIndex == 0` "invalid"

Reply:
Hmmm that's a tricky one. Is it a bug or did the devs intend that the first `bgCamData` in the `bgCamDataList` be a land-based `bgCamData` or something? I lean towards bug but idk

Reply:
I think `@bug` yeah. I could see this actually being an issue for modding